activerecord/lib/active_record/relation/finder_methods.rb

```diff
@@ -576,15 +576,24 @@ def find_last(limit)
       end
 
       def ordered_relation
-        if order_values.empty? && (implicit_order_column || !query_constraints_list.empty?)
-          # use query_constraints_list as the order clause if there is no implicit_order_column
-          # otherwise remove the implicit order column from the query constraints list if it's there
-          # and prepend it to the beginning of the list
-          order_columns = implicit_order_column.nil? ? query_constraints_list : ([implicit_order_column] | query_constraints_list)
-          order(*order_columns.map { |column| table[column].asc })
+        if order_values.empty? && (implicit_order_column || !query_constraints_list.nil? || primary_key)
+          order(*Array(_order_columns).map { |column| table[column].asc })
```

Comment:
It looks like `_order_columns` is always an array so wrap is not neccessary
```suggestion
          order(*_order_columns.map { |column| table[column].asc })
```

Reply:
Whoops, thank you!